src/core-styling/_typography-heading.scss

```diff
@@ -0,0 +1,67 @@
+@forward "typography-global";
+
+@use "sass:list";
+@use "sass:map";
+@use "settings/font";
+@use "baseline";
+@use "typography-global" as global-typography;
+
+$_h1-line-height: baseline.$unit-medium;
+$_h2-line-height: (baseline.$unit-small + baseline.$unit-smallish) / 2;
+$_default_line-height: baseline.$unit-small;
+$_font-family: font.$secondary;
+
+@mixin h1-typography() {
+  @include _typography(7, $_h1-line-height);
+}
+
+@mixin h2-typography() {
+  @include _typography(4, $_h2-line-height);
+}
+
+@mixin h3-typography() {
+  @include _typography(3, $_default_line-height);
+}
+
+@mixin h4-typography() {
+  @include _typography(2, $_default_line-height);
+}
+
+@mixin h5-typography() {
+  @include _typography(1, $_default_line-height);
+}
+
+@mixin h6-typography() {
+  @include _typography(0, $_default_line-height);
+}
+
+@mixin _typography($scale-point, $line-height) {
```

Comment:
Wonder about the order of the file (public vs private API, and selectors).

Reply:
Order matters for compilation: the mixins need to be defined before they're used in the selectors, and the local variables need to be defined before the mixin they're called from is executed.